File: CoreLib/src/main/java/org/hexnibble/corelib/commands/rc/LogRC.java
package org.hexnibble.corelib.commands.rc;

import org.hexnibble.corelib.misc.Msg;

public class LogRC extends RC {
   private final String logMessage;

   public LogRC(String logMessage) {
      super("LogRC", -1);
      this.logMessage = logMessage;
   }

   @Override
   protected void processCommand() {
      Msg.log(getClass().getSimpleName(), "", logMessage);
      setCommandStatus(COMMAND_STATUS.COMMAND_SUCCESSFULLY_COMPLETED);
   }
}

File: CoreLib/src/main/java/org/hexnibble/corelib/commands/rc/LogRC.kt
package org.hexnibble.corelib.commands.rc

import org.hexnibble.corelib.misc.Msg

/**
 * Logs a message.
 * @param logMessage The message to log.
 * @param commandID The ID of the command.
 */
class LogRC @JvmOverloads constructor(private val logMessage: String, commandID: String = "LogRC") :
  RC(commandID, -1) {
  override fun processCommand() {
    Msg.log(javaClass.simpleName, "", logMessage)
    commandStatus = COMMAND_STATUSES.COMMAND_SUCCESSFULLY_COMPLETED
  }
}